### ISSUE TITLE
This is one of: How to? 

### ISSUE THREAD

**Issue by Fortidude** [500tech/angular-tree-component#504]
Hi!

Im just curious... if there is a way to achiv something like that:

I got nodes with is:
- parent
-- child
-- child
--- another_child
--- another_child

and that works perfectly. Drag and drop, expander etc. I can drag another_child and make it child, or even parent. Cool!

Now, what I would like to achiv, is when each node have another collections, lets call it "features". And I would drag and drop features between main nodes (and still being able to drag&drop nodes)

- parent
-- features[] of parent
-- child of parent
---- features[] of child
---- another_level_child
---- another_level_child
------ features[] of another_level_child

so basically we have nodes with drag&drops and another "being" which can be drag and dropped (and expanded) between nodes.

Is there any simple way to implement this?

**Comment by adamkleingit** [comment]
Sounds like you can make `features` just another node with children, as a
transformation before giving the nodes to the tree.
So turn your data into something like this:
```
[
  {
    name: 'parent',
    children: [
      {
        name: 'features',
        type: 'feature'
        children: [ { name: 'feature1', type: 'feature' }, { name:
'feature2', type: 'feature' } ]
      },
      {
        name: 'child1',
        type: 'node'
      },
      {
        name: 'child2',
        type: 'node',
        children: [
          {
             name: 'features',
             type: 'feature'
             children: [ { name: 'feature1', type: 'feature' }, { name:
'feature2', type: 'feature' } ]
          }
        ]
      }
    ]
]
```
and then supply an `allowDrop` function that only allows dropping onto
nodes:
```
options = {
  allowDrop: (node) => node.data.type === 'node'
}
```


*Adam Klein*
*Chief Technology Officer*



*Tel.*   +972-52-7475633 <+972527475633>
*Web*  500tech.com <http://www.500tech.com/> | github.com/500tech |
@adamklein500 <http://twitter.com/adamklein500>

**Comment by adamkleingit** [comment]
Sounds like you can make `features` just another node with children, as a
transformation before giving the nodes to the tree.
So turn your data into something like this:
```
[
  {
    name: 'parent',
    children: [
      {
        name: 'features',
        type: 'feature'
        children: [ { name: 'feature1', type: 'feature' }, { name:
'feature2', type: 'feature' } ]
      },
      {
        name: 'child1',
        type: 'node'
      },
      {
        name: 'child2',
        type: 'node',
        children: [
          {
             name: 'features',
             type: 'feature'
             children: [ { name: 'feature1', type: 'feature' }, { name:
'feature2', type: 'feature' } ]
          }
        ]
      }
    ]
]
```
and then supply an `allowDrop` function that only allows dropping onto
nodes:
```
options = {
  allowDrop: (node) => node.data.type === 'node'
}
```
